apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -115,8 +119,18 @@ export const WaterfallGrid: React.FC<WaterfallGridProps> = ({
     pins,
   });
 
+  const firstActiveVersionId = activeVersionIds[0];
   const lastActiveVersionId = activeVersionIds[activeVersionIds.length - 1];
 
+  let firstActiveTaskId = "";
+  for (let i = 0; i < buildVariants[0]?.builds?.length; i++) {
+    const { builds } = buildVariants[0];
+    if (builds[i].tasks.length > 0) {
+      firstActiveTaskId = builds[i].tasks[0].id;
+      break;
+    }
+  }
+
```

Comment:
Would it perhaps make more sense to calculate this in the BuildRow component if `isFirstBuild` is `true`?